vor/elasticsearch.py

```diff
@@ -2,16 +2,50 @@
 ElasticSearch stats poller that sends statistics to Graphite.
 """
 
-import simplejson
 import time
 
 from twisted.application import service
 from twisted.internet import task
+from twisted.internet.ssl import CertificateOptions
 from twisted.python import log
-from twisted.web.client import getPage
+from twisted.web.iweb import IPolicyForHTTPS
+
+import treq
+
+from zope.interface import implementer
 
 from vor.graphite import sanitizeMetricElement
 
+@implementer(IPolicyForHTTPS)
+class _NoVerifyContextFactory(object):
+    """
+    Context that doesn't verify SSL connections.
+    """
+    def creatorForNetloc(self, hostname, port):
+        return CertificateOptions(verify=False)
+
+
+def _noVerifyAgent( *args, **kwargs ):
+    """Agent that suppresses TLS verification
+
+    treq.get(..., agent=_noVerifyAgent())
+
+    Returns the same agent every time...
+    """
+    if getattr(_noVerifyAgent, 'agent',None) is None:
```

Comment:
Can you explain why you are setting the agent like this?

Reply:
It's basically making sure that on subsequent calls, the same agent is reused. I suppose with just a limited amount of these services this might be a bit of premature optimization, but it was taken from a code example on how to disable cert verification. The upshot is that it also uses the same connection pool, so it theoretically can reuse connections between them and on each iteration.